docs/snippets/vault-gcp-store.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: external-secrets.io/v1beta1
+kind: SecretStore
+metadata:
+  name: vault-backend-gcp
+spec:
+  provider:
+    vault:
+      server: "http://my.vault.server:8200"
+      path: secret
+      version: v2
+      namespace: <vault_namespace>
+      auth:
+        gcp:
+          # Role to assume when authenticating.
+          # See https://developer.hashicorp.com/vault/docs/auth/gcp#configuration
+          role: gcp-role
+          # Subject that should be used to sign JWTs that are presented to
+          # Vault. The identity ESO is running under needs the permission
+          # `iam.serviceaccounts.signJwt` on this subject. It is included in
+          # the role `roles/iam.serviceAccountTokenCreator`.
+          serviceAccountEmail: sa@project.iam.gserviceaccount.com
```

Comment:
Does this configuration allows for using workload identity bound to other KSA, like the `gcp` provider supports?

I think from a provider point of view, we want users to be able to specify multiple secret stores, each bound to a KSA linked to a GSA with different permission set. From what I see here, the current approach only allows for ESO KSA to be used (and we try to avoid only giving the controller itself any provider credentials).

Reply:
Yes that's true.

I've updated the PR to allow using service account ref for the vault provider as well. Getting this to work for Fleet Workload Identity and making the code shared between the generator/provider required some juggling, sorry for the diff noise 😁.